rpc.go

```diff
@@ -91,6 +91,7 @@ func NewTopic(ctx context.Context, ps *pubsub.PubSub, host peer.ID, topic string
 	}
 	t.resTopic, err = newTopic(ctx, ps, host, responseTopic(topic, host), true)
 	if err != nil {
+		t.cancel()
```

Comment:
drive-by. cleaning up goroutines in case of error.

Reply:
should this be `t.Close()`?

Reply:
Right... fixed. also added some lines to properly close resources in `newTopic` https://github.com/textileio/go-libp2p-pubsub-rpc/pull/13/commits/ec807b7e2f4383141a1aa3d3da0f1fcbe6c6e5d4